tests/integration/components/menu-test.js

```diff
@@ -708,63 +862,337 @@ module('Integration | Component | <Menu>', (hooks) => {
-            'keydown',
-            Keys.ArrowDown
-          );
-        } catch (err) {
-          assert.ok(
-            err.message.startsWith('Can not `triggerKeyEvent` on disabled')
-          );
-        }
+        // Verify it is open
+        assertOpenMenuButton('[data-test-menu-button]');
+
+        // Activate first menu item
+        await triggerEvent('[data-test-item-a]', 'mouseover');
 
+        // Close menu and invoke the item
+        await triggerKeyEvent('[data-test-item-a]', 'keydown', Keys.Space);
+
+        // Verify it is closed
         assertClosedMenuButton('[data-test-menu-button]');
+
+        // Verify the "click" ent through on the `a` tag
+        assert.dom(itemClicked).hasText('Item A');
+
+        // Verify the button is focused again
+        await assertActiveElement(getMenuButton('[data-test-menu-button]'));
       });
+    });
 
-      test('it should be possible to use ArrowDown to navigate the menu items', async function (assert) {
+    module('`Escape` key', () => {
+      test('it should be possible to close the menu with Escape', async function () {
         await render(hbs`
-        <Menu as |menu|>
-          <menu.Button data-test-menu-button>Trigger</menu.Button>
-          <menu.Button data-test-menu-button >Trigger</menu.Button>
-          <menu.Items data-test-menu-items as |items|>
-            <items.Item as |item|>
-              <item.Element>
-                Item A
-              </item.Element>
-            </items.Item>
-            <items.Item as |item|>
-              <item.Element>
-                Item B
-              </item.Element>
-            </items.Item>
-            <items.Item as |item|>
-              <item.Element>
-                Item C
-              </item.Element>
-            </items.Item>
-          </menu.Items>
-        </Menu>
-      `);
+          <Menu as |menu|>
+            <menu.Button data-test-menu-button>Trigger</menu.Button>
+            <menu.Items data-test-menu-items as |items|>
+              <items.Item as |item|>
+                <item.Element>
+                  Item A
+                </item.Element>
+              </items.Item>
+              <items.Item as |item|>
+                <item.Element>
+                  Item B
+                </item.Element>
+              </items.Item>
+              <items.Item as |item|>
+                <item.Element>
+                  Item C
+                </item.Element>
+              </items.Item>
+            </menu.Items>
+          </Menu>
+        `);
+
+        assertClosedMenuButton('[data-test-menu-button]');
+
+        // Open menu
+        await triggerKeyEvent('[data-test-menu-button]', 'keydown', Keys.Space);
+
+        // Verify it is open
+        assertOpenMenuButton('[data-test-menu-button]');
 
+        // Close menu
         await triggerKeyEvent(
           '[data-test-menu-button]',
           'keydown',
-          Keys.ArrowDown
+          Keys.Escape
```

Comment:
Here we send a key event with escape. Then we assert that menu should be closed. Why is that ? I didn't notice any handlers for escape key, neither in `menu-button` nor `menu-items`. How it works ? 🤔

Reply:
when I try to execute tests locally, with commented out code:
```js
next(() => {
      if (this.buttonElement !== null) {
        this.buttonElement.focus();
      }
})
```
I see that 2 tests are failing, one of them is with escape:

<img width="921" alt="Captura de pantalla 2021-10-06 a las 22 14 31" src="https://user-images.githubusercontent.com/11621383/136276685-96631ac4-b3ee-4319-8441-2679d630b87b.png">

I suppose that it doesn't work because we don't handle `escape` properly in component. So my guess is that we need some work there.